engine/access/rpc/backend/backend_scripts.go

```diff
@@ -159,6 +144,35 @@ func (b *backendScripts) executeScriptOnExecutor(
 	}
 }
 
+func (b *backendScripts) compareScriptExecutionResults(
+	execNodeResult []byte,
+	execErr error,
+	archiveResult []byte,
+	archiveErr error,
+	blockID flow.Identifier,
+	insecureScriptHash [16]byte,
+) {
+	if bytes.Equal(execNodeResult, archiveResult) {
+		b.metrics.ScriptExecutionResultMatch()
+		b.logScriptExecutionComparison(blockID, insecureScriptHash,
+			"script execution results on Archive node and EN are equal")
+	} else {
+		b.metrics.ScriptExecutionResultMismatch()
+		b.logScriptExecutionComparison(blockID, insecureScriptHash,
+			"script execution results on Archive node and EN are not equal")
+	}
+
+	if execErr == archiveErr {
+		b.metrics.ScriptExecutionErrorMatch()
+		b.logScriptExecutionComparison(blockID, insecureScriptHash,
+			"cadence errors on Archive node and EN are equal")
+	} else {
+		b.metrics.ScriptExecutionErrorMismatch()
+		b.logScriptExecutionComparison(blockID, insecureScriptHash,
```

Comment:
```suggestion
		b.logScriptExecutionComparison(blockID, insecureScriptHash,
		return
```
we don't need to compare the result if the error is different, because the error might be different, but the result might both be `nil`, and we don't want to double count the Mismatch